### DISCUSSION TITLE
Create draft PRs on artifact update failure

### DISCUSSION THREAD

**Discussion by colinodell** [renovatebot/renovate#15121] [Suggest an Idea]
Sometimes Renovate will fail to update an artifact (like a lock file) due to dependency conflicts:

![image](https://user-images.githubusercontent.com/202034/163464565-8ee2a3c4-94a0-4d3a-b478-33432b56d0b9.png)

A PR is still opened to inform the user (which is great) but that PR itself is mergeable (not so great).  And if the CI pipeline isn't validating that all artifacts are present and in-sync it's **very** easy for a user to not understand the problem and merge the PR anyway.  (This has happened numerous times on several repositories I co-mantain.)

Merging a "bad" PR is particularly problematic with Composer: `composer install` commands will still install dependencies without issue, but attempts to manually add, remove, or updates dependencies will fail with no easy resolution (besides finding the offending merge commits and reverting them).

**I therefore propose that any PRs with artifact update problems be moved into a `Draft` state** if the platform supports it, and moved out of `Draft` if/when the issue is resolved.  This could either be a default behavior or configurable with an option like `"draftPROnFailure": true`.

Doing this will add an extra safeguard for users who miss the `You probably do not want to merge this PR as-is` message and want to avoid getting bitten by this.

**Comment by rarkins** [comment]
There should always be a renovate/artifacts failed status check in such cases. Are you seeing that, and your users are ignoring both the comment as well as the failing status check?

**Reply by colinodell** [reply]
Ah, thanks for sharing that! It looks like many of the PRs did have the failed status check, which is awesome!  Unfortunately, users sometimes ignored both the comment and status check and clicked on "update branch" ...

![image](https://user-images.githubusercontent.com/202034/163715915-7a51cc23-ca37-47bb-a2cb-7a4885a64cbf.png)

... in an attempt to "fix" the failed check.  They saw the failed check "go away" (because of the new merge commit they added), assumed all was fine, and merged the PR 😰 

Because the status check feature exists and works well, I'm fine disregarding this suggestion and I'll focus on better educating my users and adding some extra validation to my pipeline as an extra layer of defense.

Appreciate the explanation!

**Reply by rarkins** [reply]
FYI it would be good to tell them to tick the rebase/retry option as that will both update the branch and retry the artifacts update
